### PR TITLE
Base item collection unit tests

### PR DESCRIPTION
**Pull request by vigbk**

### CODE REVIEW

**Comment by vigbk** [comment]
Will add the test for get item after you merge @indradeep

**Comment by indradeep** [comment]
There is a PR that adds support for __getitem__.  However that is not merged to master yet.  I suppose we the order of merge will determine who adds the test cases.  For now this is good.

File: tests/core/base_item_collection_test.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    This class is to test abstract behavior, and thus, adds no functionality
+    """
+
+    pass
+
+
+def test_evaluate_needs_evaluation_false(
+        collection_schema_spec: Dict[str, Any]) -> None:
+    schema_loader = SchemaLoader()
+    collection_schema_spec['When'] = 'False'
```
Comment (prashastk) [inline tests/core/base_item_collection_test.py]: Lets also write one test where 'When' evaluates to an runtime error.

Comment (vigbk) [reply tests/core/base_item_collection_test.py]: Sure

**Pull request merged**